modules/c++/mem/include/mem/SharedPtr.h

```diff
@@ -144,17 +144,14 @@ class SharedPtr
         return mRefCtr->get();
     }
 
-    void reset(T* ptr = NULL)
+    void reset(std::auto_ptr<T> scopedPtr)
     {
-        // We take ownership of the pointer no matter what, so temporarily
-        // wrap it in an auto_ptr in case creating the atomic counter throws.
         // NOTE: We need to create newRefCtr on the side before decrementing
         //       mRefCtr. This way, we can provide the strong exception
         //       guarantee (i.e. the operation either succeeds or throws - the
         //       underlying object is always in a good state).
-        std::auto_ptr<T> scopedPtr(ptr);
         sys::AtomicCounter* const newRefCtr = new sys::AtomicCounter(1);
-        scopedPtr.release();
+        T* ptr = scopedPtr.release();
```

Comment:
Rather than doing this, you can say `mPtr = scopedPtr.release()` below right?  That makes really explicit the point at which ownership is being handed over.